### PR TITLE
Remove GitHub automation requiring reviews when the SDK version changes

### PR DESCRIPTION
**Pull request by jeffhandley**
Contributes to #71275 by removing the FabricBot automation that requires PR reviews when the .NET SDK version is changed. The second commit in this PR also removes 2 _disabled_ automation rules that were previously using the same capability of requesting changes to pull requests.

/cc @arkalyanms @wtgodbe @mkArtakMSFT @jaredpar

### CODE REVIEW

File: .github/fabricbot.json
Comment (jjonescz) [inline .github/fabricbot.json:1]: Can we disable auto merge so we don't end up with SDK version we don't want?

Comment (wtgodbe) [reply .github/fabricbot.json:1]: Do you have the dotnet-maestro bot account configured to do auto-merges in this repo? If not, how do you do auto-merges?

Comment (jjonescz) [reply .github/fabricbot.json:1]: > Do you have the dotnet-maestro bot account configured to do auto-merges in this repo?

Yes, I think we do. Example: https://github.com/dotnet/roslyn/pull/71259

**Comment by jeffhandley** [comment]
This was supplanted by #72305.

**Pull request closed without merging**